test/datalog_tests/server_api/tests/realization_tests.rs

```diff
@@ -0,0 +1,154 @@
+            Ok(())
+        );
+        // Check that realization has an accumulator for the sink and that the
+        // subscription exists.
+        assert!(realization.accumulator_is_subscribed_to_sink(rel_ids.clone(), &sink));
+
+        // Check that realization successfully removes the tcp sender sink.
+        assert_eq!(realization.remove_sink(&sink), Ok(()));
+        assert!(!realization.accumulator_is_subscribed_to_sink(rel_ids.clone(), &sink));
+
+        // Check that the sink accumulator can be successfully removed.
+        assert_eq!(
+            realization.remove_sink_accumulator(rel_ids, &mut server),
+            Ok(())
+        );
+    }
+
+    #[test]
+    fn add_and_remove_file_sink() {
+        // Set up dummy realization.
+        let mut realization = Realization::<HDDlog>::new();
+
+        let file = NamedTempFile::new().unwrap();
+        let path = file.path();
+
+        let mut rel_ids = BTreeSet::new();
+        rel_ids.insert(1);
+        rel_ids.insert(2);
+        rel_ids.insert(3);
+
+        let pathbuf = PathBuf::from(path);
+        let mut server = DDlogServer::new(None, HashMap::new());
+        let sink = Sink::File(pathbuf);
+
+        // Check that realization successfully adds file sink.
+        assert_eq!(
+            realization.add_sink(&sink, rel_ids.clone(), &mut server),
+            Ok(())
+        );
+        // Check that realization has an accumulator for the sink and that the
+        // subscription exists.
+        assert!(realization.accumulator_is_subscribed_to_sink(rel_ids.clone(), &sink));
+
+        // Check that realization successfully removes the file sink.
+        assert_eq!(realization.remove_sink(&sink), Ok(()));
+        assert!(!realization.accumulator_is_subscribed_to_sink(rel_ids.clone(), &sink));
+
+        // Check that the sink accumulator can be successfully removed.
+        assert_eq!(
+            realization.remove_sink_accumulator(rel_ids, &mut server),
+            Ok(())
+        );
+    }
+
+    #[test]
+    fn remove_sink_accumulator_prematurely() {
+        // Should error if the accumulator to be removed still has sinks.
+
+        let mut realization = Realization::<HDDlog>::new();
+
+        let file = NamedTempFile::new().unwrap();
+        let path = file.path();
+
+        let mut rel_ids = BTreeSet::new();
+        rel_ids.insert(1);
+        rel_ids.insert(2);
+        rel_ids.insert(3);
+
+        let pathbuf = PathBuf::from(path);
+        let mut server = DDlogServer::new(None, HashMap::new());
+        let sink = Sink::File(pathbuf);
+
+        // Check that the sink is added successfully.
+        assert_eq!(
+            realization.add_sink(&sink, rel_ids.clone(), &mut server),
+            Ok(())
+        );
+        // Check that removing the sink accumulator returns an error.
+        assert!(realization
+            .remove_sink_accumulator(rel_ids, &mut server)
+            .is_err());
+    }
+}
```

Comment:
These tests look good, but we should also test the dataflow through the system, e.g., that connecting a new sink to an accumulatator brings it up to a correct state.

Let's wrap up this PR and try to add more tests as a separate PR.

Reply:
Sounds good to me.